src/pymor/reductors/h2.py

```diff
@@ -555,13 +556,17 @@ class TFIRKAReductor(GenericIRKAReductor):
     Parameters
     ----------
     fom
-        The full-order |Model| with `eval_tf` and `eval_dtf` methods.
+        The full-order |Model| with a `transfer_function` attribute or a
+        |TransferFunction| with `eval_tf` and `eval_dtf` methods that should be
+        defined at least over the open right half of the complex plane.
     mu
         |Parameter values|.
     """
 
     def __init__(self, fom, mu=None):
-        assert isinstance(fom, InputOutputModel)
+        assert hasattr(fom, 'transfer_function') or isinstance(fom, TransferFunction)
+        if hasattr(fom, 'transfer_function'):
+            fom = fom.transfer_function
```

Comment:
```suggestion
        assert isinstance(fom, TransferFunction) or  hasattr(fom, 'transfer_function')
        if not isinstance(fom, TransferFunction):
            fom = fom.transfer_function
```

Seems a bit clearer to me.